tenant_users/tenants/tasks.py

```diff
@@ -43,11 +51,20 @@ def provision_tenant(tenant_name, tenant_slug, user_email):
         )
 
         # Add user as a superuser inside the tenant
-        tenant.add_user(user, is_superuser=True)
-    except:
-        if tenant is not None:
-            #Flag is set to auto-drop the schema for the tenant
-            tenant.delete(True)
-        raise
+    else:  # django-tenants
+        tenant = TenantModel.objects.create(name=tenant_name,
+                                            slug=tenant_slug,
+                                            schema_name=schema_name,
+                                            owner=user)
+
+        # Add one or more domains for the tenant
+        domain = get_tenant_domain_model().objects.create(domain=tenant_domain,
+                                                          tenant=tenant,
+                                                          is_primary=True)
+
+    tenant.add_user(user, is_superuser=True, is_staff=is_staff)
+    # if tenant is not None:
+    #     #Flag is set to auto-drop the schema for the tenant
+    #     tenant.delete(True)
```

Comment:
Have you had the tenant code fail? It shouldn't fail.
I got rid of it as a try and except without excepting an exception such as ValueError or KeyError is bad practice. If you write an error in code it will go to the exception which wouldn't be good.
If you had an error what was the exception?